server/middlewares/paramsValidator.js

```diff
@@ -8,17 +8,45 @@ import { badHttpResponse } from '../utilities/httpResponse';
  * Validation error response if param is not valid or absent | next function
  */
 const paramValidator = paramsDefinition => (request, response, next) => {
-  const { params } = request;
+  const { params, body } = request;
   const errors = {};
+  const urlParams = paramsDefinition.params;
+  const bodyParams = paramsDefinition.body;
+  let isMatch;
 
-  Object.keys(paramsDefinition).forEach((key) => {
-    const definition = paramsDefinition[key];
-    const isMatch = definition.pattern.test(params[key]);
+  if (urlParams) {
+    Object.keys(urlParams).forEach((key) => {
+      const definition = urlParams[key];
+      isMatch = definition.pattern.test(params[key]);
 
-    if (!isMatch) {
-      errors[key] = `The ${key} parameter must be a ${definition.type}`;
-    }
-  });
+      if (!isMatch) {
+        errors[key] = `The ${key} parameter provided is not a valid ${key}`;
+      }
+    });
+  }
+
+  if (bodyParams) {
```

Comment:
This is hard to follow and seems like an unnecessary complication. I'd recommend adding custom validation to what you're supposed to be working on and leaving the rest of the codebase as is

Reply:
@seyi-adeleke I initially created this function to check for url parameters but I wanted to be able to use it for the body parameters as well. I'd just leave it as it was before this pull request.